cmd/mayactl/app/command/volume_stats.go

```diff
@@ -114,53 +114,54 @@ func (c *CmdVolumeStatsOptions) RunVolumeStats(cmd *cobra.Command) error {
 	replicaStatus := strings.Split(annotations.ReplicaStatus, ",")
 	for _, repStatus = range replicaStatus {
 		if repStatus == "Pending" {
-			statusArray[replicaCount] = "Unknown"
-			statusArray[replicaCount+1] = "Unknown"
-			statusArray[replicaCount+2] = "Unknown"
-			replicaCount += 3
+			statusArray = append(statusArray, "Unknown")
+			statusArray = append(statusArray, "Unknown")
+			statusArray = append(statusArray, "Unknown")
+
 		}
 	}
 
 	replicas := strings.Split(annotations.Replicas, ",")
 	for _, replica := range replicas {
-		errCode1, err := client.GetStatus(replica+":9502", &status)
+		replicaClient := client.ReplicaClient{}
+		errCode1, err := replicaClient.GetVolumeStats(replica+":9502", &status)
 		if err != nil {
 			if errCode1 == 500 || strings.Contains(err.Error(), "EOF") {
-				statusArray[repCount] = replica
-				statusArray[repCount+1] = "Waiting"
-				statusArray[repCount+2] = "Unknown"
-
+				statusArray = append(statusArray, replica)
+				statusArray = append(statusArray, "waiting")
+				statusArray = append(statusArray, "Unknown")
```

Comment:
I really dont understand how these decisions are made.. i.e. setting waiting, unknown.
It seems that statusArray is a blackbox.

Reply:
@AmitKumarDas  yes, It's based on the error code received.